aten/src/ATen/native/native_functions.yaml

```diff
@@ -3267,12 +3267,20 @@
   dispatch:
     CompositeExplicitAutograd: linspace
 
+- func: linspace.Tensor(Tensor start, Tensor end, int steps, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor
```

Comment:
Before this change, the following seemed to work:
```
import torch
torch.linspace(torch.tensor(0), 1, 10)
```
After this change, it raises an error. We probably just need more overloads. The other overloads should be CompositeImplicitAutograd and just wrap the start/end tensor in a Tensor and call linspace.Tensor.

Reply:
Just confirmed that after the change it still works.

Reply:
Sounds good

Reply:
I end up added multiple overloads for different Tensor/Scalar input combinations because without these overload s some code complains that 'item' cannot be called inside vmap, for example.

OpInfo tests added.